lib/p2p/NodeList.ts

```diff
@@ -170,23 +170,26 @@ class NodeList extends EventEmitter {
     const node = this.nodes.get(nodePubKey);
 
     if (node) {
-      const promises: PromiseLike<any>[] = [
-        this.repository.addReputationEvent({ event, nodeId: node.id }),
-      ];
+      const promises: PromiseLike<any>[] = [];
 
       this.updateReputationScore(node, event);
 
-      if (node.reputationScore < NodeList.BAN_THRESHOLD) {
+      if (node.reputationScore < NodeList.BAN_THRESHOLD && !node.banned) {
         promises.push(this.setBanStatus(node, true));
 
-        const events = await this.repository.getReputationEvents(node);
-        this.emit('node.ban', nodePubKey, events);
-      } else if (node.banned) {
+        const negativeReputationEvents = (await this.repository.getReputationEvents(node))
```

Comment:
Maybe we can extract `getNegativeReputationEvents(events: ReputationEvents[], event: ReputationEvent) => ReputationEvents[]` from this function. This would be a pure function and easily unit testable that it:
- returns up to 10 reputation events
- outputs only negative events
- prepends current to the filtered list